### ISSUE TITLE
Classifier Issues

### ISSUE THREAD

**Issue by dodgerblue11** [ningyu1991/GANFingerprints#5]
Hello,

I am testing your code on generated images from the pre-trained models you have provided. There are quite a few issues encountered while attempted to run the classifier on this dataset we have creating (testing portion of your code). 
- The environment had to be build as CUDA 10.0, Python3.7, and tensorflow-gpu 1.13.1, as previous versions were incompatible
- once run with the pk pre-trained model provided, in file GANFingerprints/classifier/util_scripts.py, line 23 `C_im = misc.load_network_pkl(model_path)` cannot be used as is anymore when executing line 57 `logits = C_im.run(im, minibatch_size=1, num_gpus=1, out_dtype=np.float32)` as this ends up being a tuple. I ended up changing line 23 to `G, D, Gs = misc.load_network_pkl(model_path)` and using `logits = D.run(im, minibatch_size=1, num_gpus=1, out_dtype=np.float32)` to run the classifier.
- upon running the classifier, assignment of labels gets skipped over. This would mean it does not match the labels at the top. What could we use as labels if logits.shape[1] = 1?
- If I use labels as list(labels_1) or list(labels_2) , it classifies everything as from CelebA, which is incorrect as the dataset I am passing in is all fake images generated from sngan model.

Please let me know how I can fix this.
Thanks

**Comment by jiangds2018** [comment]
> Hello,
> 
> I am testing your code on generated images from the pre-trained models you have provided. There are quite a few issues encountered while attempted to run the classifier on this dataset we have creating (testing portion of your code).
> 
> * The environment had to be build as CUDA 10.0, Python3.7, and tensorflow-gpu 1.13.1, as previous versions were incompatible
> * once run with the pk pre-trained model provided, in file GANFingerprints/classifier/util_scripts.py, line 23 `C_im = misc.load_network_pkl(model_path)` cannot be used as is anymore when executing line 57 `logits = C_im.run(im, minibatch_size=1, num_gpus=1, out_dtype=np.float32)` as this ends up being a tuple. I ended up changing line 23 to `G, D, Gs = misc.load_network_pkl(model_path)` and using `logits = D.run(im, minibatch_size=1, num_gpus=1, out_dtype=np.float32)` to run the classifier.
> * upon running the classifier, assignment of labels gets skipped over. This would mean it does not match the labels at the top. What could we use as labels if logits.shape[1] = 1?
> * If I use labels as list(labels_1) or list(labels_2) , it classifies everything as from CelebA, which is incorrect as the dataset I am passing in is all fake images generated from sngan model.
> 
> Please let me know how I can fix this. Thanks

I have the same problem as you. Have you solve it?
